ops/profiles_gcp.go

```diff
@@ -0,0 +1,58 @@
+package ops
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+)
+
+var (
+	onGCEOnce sync.Once
+	onGCE     bool
+)
+
+func init() {
+	OnGCE()
+}
+
+// OnGCE reports whether this process is running on Google Compute Engine.
+func OnGCE() bool {
+	onGCEOnce.Do(initOnGCE)
+	return onGCE
+}
+
+func initOnGCE() {
+	onGCE = testOnGCE(&net.Resolver{}, 3*time.Second)
+}
+
+type resolver interface {
+	LookupHost(ctx context.Context, host string) (addrs []string, err error)
+}
+
+func testOnGCE(resolver resolver, timeout time.Duration) bool {
```

Comment:
Initially I thought we would need to detect when running on GCP, and that detection seemed flaky for k8s, so I added this function (inspired from google own libs). But then I realised we still needed to know the google_project_id for tracing, so I left that in the configuration. Left this method anyway since it should detect gcp profile more accurately..